src/StructuredLogger/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -1358,22 +1358,35 @@ private BuildEventArgs ReadPropertyReassignmentEventArgs()
             string location = ReadDeduplicatedString();
 
             string message = fields.Message;
-            if (_fileFormatVersion >= 13)
+            if (_fileFormatVersion >= 13 && _fileFormatVersion < 25)
             {
                 message = GetPropertyReassignmentMessage(propertyName, newValue, previousValue, location);
             }
+            else if (_fileFormatVersion >= 25 && !string.IsNullOrEmpty(fields.Message))
+            {
+                message = string.Format(fields.Message, propertyName, newValue, previousValue, $"{fields.File} ({fields.LineNumber},{fields.ColumnNumber})");

```

Comment:
Where is the `fields.Message` comming from?

It's not being de/serialized - correct?

Reply:
the message comes from MSBuild here.

Reply:
It comes as a raw message it order to save string optimizations on msbuild side

Reply:
If it's properly deduplicated - than all is good!